Adds in a HEAVY DUTY saw

Kmc2000 opened the pull request:
As the title says, this features a circular saw which:

Does 21 damage when on,  5 when off
Must be turned on to use all of the following:
        It can rip through walls, floors and wall girders, it's also fantastic for smashing windows!

Where it spawns:

![](https://i.gyazo.com/8c76f32d325ea8733822cd69485651c0.png)

what it looks like! 

![](https://i.gyazo.com/a5f09034d6f3ef8aff3a58b60dd0ad2e.png)

It even features its own suicide message :+1:

Why the fuck does this exist?

RCDS take a hard to get resource to do anything, and sometimes you just want to tear things down quickly.

It runs on welding fuel, and can only be refuelled when the cover is open

Features planned:

Overheating

Is it good, bad, ugly, or OP?

Discuss below
# 

On a final note....

![](https://i.gyazo.com/a8e3a66811f505fbca63ff571e05c232.png)

:cl:
rscadd: Kmc2000 added: A circle saw now spawns in engineering! fuelled by oil, it can cut and slice through tough metal.
/:cl:

Kmc2000 commented:
Special thanks to XtheDark for helping me debug it, and teaching me how to code

Jc1469 commented:
Welders take fuel. RCDs take matter, as you said.

This is just an OP tool for lazy people.  
No. andyourspriteslookhorriblegodieroxy

X-TheDark commented:
Not to mention this ASSBLAST'ing all over reinforced walls as well.

Kmc2000 commented:
It does rwalls as well.....



Nope ok yeah adminbus only AKA rip pr

FluffySurvivor commented:
Add an overheat system. 

You can only take down a few structures before it overheats and you won't be able to use it for a while after. 

and the sprite looks great :+1:

GrayRachnid commented:
remove it's ability to tear through rwalls and make it use fuel.

even saws in real life use fuel.

Kmc2000 commented:
Fuel? i think that can be arranged actually

FluffySurvivor commented:
Fuel? So it's just a reskinned rcd? Great idea

X-TheDark commented:
You can robust someones ass with this, though.

Jc1469 commented:
RCD does 10 damage when thrown. I'd say that's more than robust.

Kmc2000 commented:
Do not merge yet, adding in fuel so it's less ungodly

ShadowDeath6 commented:
maybe have it use oil or something, I think chem can make oil

Jc1469 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,60 @@
+				F.dismantle_wall()
+
+/obj/item/weapon/heavy_saw/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is cutting into their skull with the [src.name]! It looks like they're trying to do it themselves!.</span>")
```
!.

ShadowDeath6 replied:
should make it decapitate

Jc1469 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,60 @@
+				F.ChangeTurf(F.baseturf)
+
+		else if(istype(target, /turf/closed/wall))
+			user << "<span class='notice'>You start sawing through the wall plates</span>"
```
Nice full stop.

Jc1469 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,60 @@
+	if(on)
+		playsound(src.loc, 'sound/weapons/circsawhit.ogg', 50, 1)
+		if(istype(target, /turf/open/floor))
+			user << "<span class='notice'>You start sawing through the floor plates</span>"
```
Nice full stop.

Jc1469 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,60 @@
+
+/obj/item/weapon/heavy_saw
+	name = "Industrial Saw"
+	desc = "A heavy duty saw used for cutting through hull plates during search and rescue operations"
```
Nice full stop.

Kmc2000 commented:
ADDED:


Fuel, allowing refuels, and the user is now able to check fuel levels

It holds 100 units, 10 fuel is used per action, it may only be refuelled when turned off

Removed ability to cut through reinforced walls


This should be good to go, not seeing any errors when running locally

Kmc2000 commented:
To clarify, it takes welding fuel, so it can be refuelled at fuel tanks

Jc1469 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -601,20 +604,69 @@
+		user <<"<span class='warning'>Error, not enough fuel</span>"
 
 /obj/item/weapon/heavy_saw/suicide_act(mob/user)
 	user.visible_message("<span class='suicide'>[user] is cutting into their skull with the [src.name]! It looks like they're trying to do it themselves!.</span>")
```
!. still

X-TheDark commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+	attack_verb = list("attacked", "slashed", "eviscerated", "cut")
+	var/on = FALSE
+	var/max_fuel = 100
+	var/currentfuel = 100
```
You don't need another var to keep track of fuel, since you are already using reagents.

Kmc2000 replied:
I used to as i couldn't get the loop to run with get_fuel or the regent contents, it wanted an actual number

If you have a less hacky fix, do tell me

Super3222 replied:
Remove it.

ParagonPrime commented:
>saw that can easily cut through metal walls and plating
>same damage as a fine saw used for surgery

make it able to dismember people's limbs with 100% chance but make it two-handed and take 4-5 seconds so no taser combo cafe

Notamaniac commented:
Blade needs more shading, right now it just looks like grey cardboard. Shade a sector of the saw a lighter grey, or just look up a circular saw on google and match that shading. You've also got a weird pixel on the bottom right of the blade that doesn't seem like it should be there.

I'm also a little eeeeeehhh on the colouring in general, I'm not sure whether it's my pure black trigger (black seems to be really overpowering here), or if it's something that I just can't put my finger on.

Kmc2000 commented:
>make it able to dismember people's limbs with 100% chance but make it two-handed and take 4-5 seconds so no taser combo cafe

Upping the damage.... i'll let others discuss that one right there, however it is sharp, so it has the chance to dismember, but an INSTA GG DISMEMBERINO SAW seems really OTT

Super3222 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+	else
+		if(M)
+			M << "<span class='warning'>You need more welding fuel to complete this task!</span>"
+		return 0
```
You don't need to return 0 here, unless you are going to use it for something else.

Super3222 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+					var/turf/closed/wall/F = target
+					if(do_after(user, 100, target = target))
+						F.dismantle_wall()
+						remove_fuel(10)
```
You should carry the user in the second parameter, otherwise it'll never tell them whether their fuel is out.

Super3222 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+				var/turf/open/floor/F = target
+				if(do_after(user, 50, target = target))
+					F.ChangeTurf(F.baseturf)
+					remove_fuel(10)
```
You should carry the user in the second parameter, otherwise it'll never tell them whether their fuel is out.

Super3222 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+			playsound(src.loc, 'sound/effects/refill.ogg', 50, 1, -6)
+			currentfuel = 100
+			return
+	if(currentfuel == 0)
```
`if(reagents.get_reagent_amount("welding_fuel") <= 0`

Super3222 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+		user <<"<span class='warning'>Error, not enough fuel</span>"
+
+/obj/item/weapon/heavy_saw/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is cutting into their skull with the [src.name]! It looks like they're trying to do it themselves!.</span>")
```
Make it auto decapitate their head please

Super3222 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -561,3 +561,112 @@
+			currentfuel = 100
+			return
+	if(currentfuel == 0)
+		user <<"<span class='warning'>Error, not enough fuel</span>"
```
Error! Not enough fuel.

Super3222 commented:
Other Ideas:
- Make a screwdriver open the maintenance of this weapon for transferring welder fuel inside. Otherwise it whys it takes a moment to saw the welder fuel tank and than explodes.
- Make this weapon do 21 brute damage like a chainsaw (yes, the ones you can craft any time, and any place). Let's say ... 9 brute damage when not on?
- Make it blind people when using, promoting use of welder helmets.
- Make this weapon slow you down, so people can't run around with it. `slowdown = 2` or something. (slodown 2 makes it just as heavy as a flag in CTF)
- I can use a welder 20 times before it runs out and, with this weapon I can only use it 10 times.
- Up the maxfuel to 400.
- Maybe make a tiny bit of sparks generate every time it's used on a turf?
- Add an overheat system where keeping it on and using it repeatedly will up the numbers of a variable. If attacked (by picking up) or ever used, make it use another proccall which checks for the type of gloves they're wearing. Depending on the gauge's number, it may completely burn someones hands with no gloves and make them drop the weapon. Only black gloves keep you from being burned. You'd probably need to lower the overheat gauge in a `/process()` but, remember that process runs quickly (every 1-2 seconds if I'm not mistaken).

Super3222 commented:
Other Ideas:
- Make a screwdriver open the maintenance of this weapon for transferring welder fuel inside. Otherwise it whys it takes a moment to saw the welder fuel tank and than explodes.
- Make this weapon do 21 brute damage like a chainsaw (yes, the ones you can craft any time, and any place). Let's say ... 9 brute damage when not on?
- Make it blind people when using, promoting use of welder helmets.
- Make this weapon slow you down, so people can't run around with it. `slowdown = 2` or something. (slowdown 2 makes it just as heavy as a flag in CTF)
- I can use a welder 20 times before it runs out and, with this weapon I can only use it 10 times.
- Up the maxfuel to 400.
- Maybe make a tiny bit of sparks generate every time it's used on a turf?
- Add an overheat system where keeping it on and using it repeatedly will up the numbers of a variable. If attacked (by picking up) or ever used, make it use another proccall which checks for the type of gloves they're wearing. Depending on the gauge's number, it may completely burn someones hands with no gloves and make them drop the weapon. Only black gloves keep you from being burned. You'd probably need to lower the overheat gauge in a `/process()` but, remember that process runs quickly (every 1-2 seconds if I'm not mistaken).

GrayRachnid commented:
Super, I agree with most of your ideas except this

>I can use a welder 20 times before it runs out and, with this weapon I can only use it 10 times.
Up the maxfuel to 400.

That would mean they can take down apprximately a lot of walls, and that's tearing through them quickly, without the need to refuel. a lot is a big number, and they should need to refuel every now and then.

Kmc2000 commented:
Oh boy, this is getting pretty complicated now, i'll see what i can do to get those ideas into the design

Super3222 commented:
I see what you mean. What I had in mind was using it 45 times, so it's in between industrial (45 uses) and the upgraded version (80 uses). Keep in mind that these are NOT faster than actual welders, Gray. These go through the same amount of time it takes a welder to tear down a wall.

I'm trying to get it to be somewhat better than a welder but, still something what you'd only want to use in emergencies due to it's speed and terrible flexibility.

More Ideas:

Solution A: If you make it faster, keep the max fuel at 100. This is my preferable solution.
Solution B: If you up the max amount of fuel (not going over 150), make it cut down walls incredibly slow.

With solution B, add this under it's defines `toolspeed = 0.5`.
Also instead of `if(do_after(user, 100, target = target))` swap it out with`if(do_after(user, slicing_duration/toolspeed, target = target))`.
And include a slicing duration variable for open floors, then do the same thing.
Also make it drain fuel by keeping it on or something. That'd be optional though.

Jc1469 commented:
make retarded and op like everyone wants and make it never be used ever by not putting it on the map.
And holy fuck fix the !. on the suicide line.

AdamElTablawy commented:
Bad idea. RCD's exist for a reason-this is just a snowflake item with a shitty sprite and basically a roundstart sonic jackhammer that doesn't fuck up rwalls.

Also I am against adding more 15 damage items especially one as easy access as this.

Altogether a reskinned, snowflake RCD with easier to come by ammo. No.

If you want to add something, add a floor tile replacer that replaces floor tiles with whatever you have loaded-crowbarring up not required.

AdamElTablawy commented:
Also what the fuck is this code

Kmc2000 commented:
Do not merge, adding in everyone's suggestions

fizzle7 commented:
>Fuel

Saws use electricity, give it a battery that has to be changed with a screwdriver.

Kmc2000 commented:
>Saws use electricity, give it a battery that has to be changed with a screwdriver.

Not all saws do you know....


Also then we'd lost the chainsaw sounds on it...

fizzle7 commented:
Why would you use a fuel saw that's from like 20 centuries ago in a 26th century space station?

Kmc2000 commented:
>Why would you use a fuel saw that's from like 20 centuries ago in a 26th century space station?
Because nt are cheap bastards

fizzle7 commented:
:sigh:

AdamElTablawy commented:
Make it use a highcap or supercap cell, and consume significant charge every use. Disable the ability for it to cut walls apart, but allow it to cut girders and rgirders as it can now, and destroy wooden structures and tables in one click. Allow it to slice apart wooden walls or doors quickly. Allow it to slice airlock bolts for significant charge and taking longer than an esword to do so. Make it bulky so it cannot fit in a backpack, make sure it doesn't fit in belt or back slots, and make it slow you down if carried. Should do 19 damage when on and drain some blood similar to syndie hemmoharrage shots-like 5% blood removed onhit.

It's now good at slicing apart girders and wooden constructions, remains one of the only ways to remove floor plating, can disable airlock bolts with effort, and isn't a bad weapon, but slows you down, is unwieldy, and needs something rarer than fuel to recharge.

Kmc2000 commented:
@AdamElTablawy  i like those ideas, the powercell thing is going to fuck up everything though...you sure fuel isn't gonna work out? ((spent hours getting that to work, grumble))

As for walls, i don't see the point of removing the ability to slice through them TBH, that's one of its main purposes.


can everyone discuss adam's ideas, to see if i'm gonna impliment them

AdamElTablawy commented:
Wall slicing is like a sonic jackhammer/ rcd. It shouldn't be an upgrade to the rcd-which it would be if it killed walls too.

Welding fuel is too easy to come by.

Three options-one, power cell.

Two-inbuilt power cell and let it recharge in a recharger

Three-require oil from chemistry.

AdamElTablawy commented:
Also cutting holes in the floor better freaking log.

Kmc2000 commented:
Alright, oil sounds good, that means that i don't have to change everything, will nerf speed at which it cuts the floors up to like...15 seconds

AdamElTablawy commented:
15 seconds seems to be too long for it to be useful. 5 would work.

Kmc2000 commented:
Floor speed altered, walls take 7 seconds as of now.


Working on making it use oil and burn your hands when it overheats

AdamElTablawy commented:
Hand burn is unnecessary with the nerfs I suggested. Just make it spark so if you use it in a plasma fire you become toast.

Alternatively, black or combat gloves should stop burning hands.

Kmc2000 commented:
It already sparks whenever you cut anything with it

Kmc2000 commented:
What it does now:

Takes oil to run

Cuts floor plates in  5 seconds, walls in 7 

Can be refuelled with any container that has oil in it


Tank can be removed by screwdrivering it open, then crowbaring it, to remove any unwanted chemicals



I decided to keep the wall sawing in, otherwise it's a bit naff

fizzle7 commented:
> Takes oil to run

What is this? A 1920s vision of robots in the future that chug oil? Oil is for lubrication not fuel.

Silk-Touch commented:
Suicide message should be that they are trying to cut themselves in half and should dismember everything.

Kmc2000 commented:
Do not merge again, more bugs

Kmc2000 commented:
Good to go lads

Kmc2000 commented:
Special thanks to:


XtheDark for teaching me the basics, and helping me debug 
Super3222 for help with debugging, the suicide, making it less shit and more balanced


_thanks guys_

Kmc2000 commented:
This is now merge-ready

ParagonPrime commented:
this seems incredibly underpowered

i guarantee it's going to be extremely underused except for creating breaches by griefers or antags

Kmc2000 commented:
>this seems incredibly underpowered 

Explain, it's one of the few things that can rip through floors, it tears down windows quickly and is a robust weapon

Jc1469 commented:
[4:05 PM] Johncena1469: It's basically
[4:05 PM] Johncena1469: A welder capable of destroying floors, which does 21 damage.

ParagonPrime commented:
*A welder that slows you down heavily and takes up a shitton of space

AdamElTablawy commented:
Exactly as planned-the item is useless.

Now close the PR and go make an auto-floor-tile replacer.

ShadowDeath6 commented:
>auto floor tile replacer
floorbots not being a thing

remove adam

Super3222 commented:
The suicide was my favorite part.

ShadowDeath6 commented:
make the suicide decapattack when

Kmc2000 commented:
remove adam

ShadowDeath6 commented:
fix ur fukn conflix

Alek2ander commented:
Given that the RCD is not a unique item, and this is, this should be a steal objective.

Kmc2000 commented:
I fixed the fuckers, yet they popped straight back up...hnngh

Kmc2000 commented:
DNM just yet

ShadowDeath6 commented:
inhands

Super3222 commented:
this pr stood beautifully at 69 replies and you blokes ruined it
anyway kmc, thought you were ditching this.

Kmc2000 commented:
Succ.


Some people evidently want it, so i'll necrocode it

Dagdammit commented:
I've lost track of where this stands in terms of features- can it saw through door bolts? That sounds rad. A two-handed saw that can cut regular walls, floor plating, annhilates windows, and has enough fuel for 10 uses at a time sounds fine. I don't care about huge damage.

Handheld saws can cut through metal plating just fine, we do it all the time at my heavy industry job. They don't run on liquid fuel but neither do IRL welders, it's fine. (cutting through reinforced walls, THAT would be pretty damn OP.)

If it's potent enough (say, still also wrecks doors) you could help balance it by spawning it in Secure Storage instead.

Super3222 commented:
@Kmc2000 If you're planning to work on this in the future, than can I close this PR?
